research/src/fetch_sex_places.py

```diff
@@ -242,6 +243,7 @@ def parse(self, keyword, content):
         :param place: dictionary with id key.
         :return: dictionary updated with name, address and phone.
         """
+        global url  # url is define in the installation process
```

Comment:
We don't need global `url` here. It's only used for logging and there (line 255) the proper value would be `self.url.details(place_id)`. Maybe define in this line (246) `url = self.url.details(place_id)` and then user the `url` variable in lines 253 and 255.